app/models/request.rb

```diff
@@ -40,6 +44,26 @@ def create_child
     end
   end
 
+  def root
+    root? ? self : parent
+  end
+
+  def root?
+    parent_id.nil?
+  end
+
+  def leaf?
+    number_of_children.zero?
+  end
+
+  def pure_leaf?
```

Comment:
@bzwei Should this be called a branch or a subtree? leaf and pure_leaf are confusing.